service/history/commandChecker.go

```diff
@@ -191,23 +191,66 @@ func (c *workflowSizeChecker) checkIfNumChildWorkflowsExceedsLimit() error {
 		tag.WorkflowRunID(key.RunID),
 	)
 
-	numPending := len(c.mutableState.GetPendingChildExecutionInfos())
-	errLimit := c.numPendingChildExecutionsLimit
 	if withinLimit(numPending, errLimit) {
 		return nil
 	}
-	c.metricsHandler.Counter(metrics.NumPendingChildWorkflowsTooHigh.GetMetricName()).Record(1)
+	c.metricsHandler.Counter(metricName).Record(1)
 	err := fmt.Errorf(
-		"the number of pending child workflow executions, %d, "+
-			"has reached the error limit of %d established with %q",
+		"the number of %s, %d, for run %q of workflow %q in namespace %q has reached the error limit of %d",
+		resourceName,
 		numPending,
+		key.RunID,
+		key.WorkflowID,
```

Comment:
I don't mean to have the workflow id and run id in the error. Can we log an error message? Or the caller will do it?

Reply:
We do that on line 207, and the logger is tagged with those values on 187

Reply:
I've removed this from the inline error message